service/authorization/authorization.go

```diff
@@ -287,6 +295,11 @@ func (as *AuthorizationService) getDecisions(ctx context.Context, dr *authorizat
 
 	for raIdx, ra := range dr.GetResourceAttributes() {
 		for ecIdx, ec := range dr.GetEntityChains() {
+			// check if we already have a decision for this entity chain
+			responseIdx := (raIdx * len(dr.GetEntityChains())) + ecIdx
+			if response[responseIdx] != nil {
```

Comment:
It's not obvious to me what condition would result in a duplicate.  As long as an overflow does not occur, and `dr.GetEntityChains()` remains constant, I don't believe a duplicate responseIdx is possible.

Is there a condition I am missing?